lib/ddtrace/profiling/recorder.rb

```diff
@@ -14,9 +14,10 @@ def initialize(event_classes, max_size)
         end
       end
 
-      def push(event)
-        raise UnknownEventError, event.class unless @buffers.key?(event.class)
-        @buffers[event.class].push(event)
+      def push(events)
+        event_class = events.is_a?(Array) ? events.first.class : events.class
+        raise UnknownEventError, event_class unless @buffers.key?(event_class)
```

Comment:
This assumes all events in the Array are of the same class.

Reply:
👍 I made the same assumption and limitation in the Python recorder — I think it's a good trade-off.